sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -2794,4 +2794,35 @@ class SubquerySuite extends QueryTest
       checkAnswer(df3, Row(7))
     }
   }
+
+  test("SPARK-49646: handle under/setOp under limit/aggregate with predicates cannot pulled up") {
```

Comment:
can we move the test to `sql/core/src/test/resources/sql-tests/inputs/join-lateral.sql`?